WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingManager.java

```diff
@@ -90,4 +107,112 @@ private void addData(Container c, User user, String schemaName, String tableName
             throw new RuntimeException(e);
         }
     }
+
+    public void submitRequestForm(User user, Container container, WNPRC_PurchasingController.RequestForm requestForm)
+    {
+        UserSchema us = QueryService.get().getUserSchema(user, container, "ehr_purchasing");
+        String requestId = UUID.randomUUID().toString().toUpperCase();
+        Map<String, Object> row;
+
+        //New vendor data
+        List<Map<String, Object>> newVendorData = new ArrayList<>();
+        if (requestForm.getHasNewVendor())
+        {
+            row = new CaseInsensitiveHashMap<>();
+            row.put("vendorName", requestForm.getNewVendorName());
+            row.put("streetAddress", requestForm.getNewVendorStreetAddress());
+            row.put("city", requestForm.getNewVendorCity());
+            row.put("state", requestForm.getNewVendorState());
+            row.put("country", requestForm.getNewVendorCountry());
+            row.put("zip", requestForm.getNewVendorZip());
+            row.put("phoneNumber", requestForm.getNewVendorPhoneNumber());
+            row.put("faxNumber", requestForm.getNewVendorFaxNumber());
+            row.put("email", requestForm.getNewVendorEmail());
+            row.put("url", requestForm.getNewVendorUrl());
+            row.put("notes", requestForm.getNewVendorNotes());
+            //TODO: set qc state - need to verify workflow with client
+            newVendorData.add(row);
+        }
+
+        //Purchasing request data
+        List<Map<String, Object>> purchasingRequestsData = new ArrayList<>();
+
+        row = new CaseInsensitiveHashMap<>();
+        row.put("requestId", requestId);
+        row.put("vendorId", requestForm.getVendor());
+        row.put("account", requestForm.getAccount());
+        row.put("otherAcctAndInves", requestForm.getAccountOther());
+        row.put("shippingInfo", requestForm.getShippingDestination());
+        row.put("justification", requestForm.getPurpose());
+        row.put("shippingAttentionTo", requestForm.getDeliveryAttentionTo());
+        row.put("comments", requestForm.getComments());
+        row.put("assignedTo", user.getUserId()); //TODO : this will be set to purchasing admin
+        row.put("qcState", requestForm.getQcState());
+        purchasingRequestsData.add(row);
+
+        //Line items data
+        List<Map<String, Object>> lineItemsData = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        WNPRC_PurchasingController.LineItem[] lineItems = mapper.convertValue(requestForm.getLineItems(), WNPRC_PurchasingController.LineItem[].class);
```

Comment:
This seems like a lot of gyrations from receiving JSON -> Java objects -> Map<String, Object>. Can we just feed the JSON objects (which should be easy to get as maps) directly into the QueryUpdateService calls?

Reply:
Agreed! now changed!